app/models/katello/rhel_lifecycle_status.rb

```diff
@@ -0,0 +1,169 @@
+      return MAINTENANCE_SUPPORT if Date.today <= RHEL_EOS_SCHEDULES[release]['maintenance_support']
+      return APPROACHING_END_OF_SUPPORT if warn_date.present? && Date.today <= warn_date
+      return EXTENDED_SUPPORT if Date.today <= RHEL_EOS_SCHEDULES[release]['extended_support']
+      return SUPPORT_ENDED
+    end
+
+    def self.status_name
+      N_('RHEL lifecycle')
+    end
+
+    def self.humanized_name
+      'rhel_lifecycle'
+    end
+
+    def self.full_support_end_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('full_support')
+    end
+
+    def self.maintenance_support_end_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('maintenance_support')
+    end
+
+    def self.extended_support_end_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('extended_support')
+    end
+
+    def self.approaching_end_of_support_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      eos_date(eos_schedule: eos_schedule) - EOS_WARNING_THRESHOLD
+    end
+
+    def self.eos_date(eos_schedule: nil)
+      return nil unless eos_schedule
+      RHEL_EOS_SCHEDULES[eos_schedule]&.[]('extended_support') ||
+        RHEL_EOS_SCHEDULES[eos_schedule]&.[]('maintenance_support')
+    end
+
+    def self.to_label(status, eos_date: nil)
+      case status
+      when FULL_SUPPORT
+        N_('Full support')
+      when MAINTENANCE_SUPPORT
+        N_('Maintenance support')
+      when EXTENDED_SUPPORT
+        N_('Extended support')
+      when APPROACHING_END_OF_SUPPORT
+        if eos_date.present?
+          N_('Approaching end of support (%s)') % eos_date.strftime('%Y-%m-%d')
+        else
+          N_('Approaching end of support')
+        end
+      when SUPPORT_ENDED
+        N_('Support ended')
+      else
+        N_('Unknown')
+      end
+    end
+
+    def to_label(_options = {})
+      self.class.to_label(status, eos_date: eos_date)
+    end
+
+    def eos_date
+      self.class.eos_date(eos_schedule: rhel_eos_schedule)
+    end
+
+    def warn_date
+      eos_date - EOS_WARNING_THRESHOLD
+    end
+
+    def rhel_eos_schedule
+      host&.operatingsystem&.rhel_eos_schedule
+    end
+
+    def to_global(_options = {})
+      if [FULL_SUPPORT, MAINTENANCE_SUPPORT, EXTENDED_SUPPORT].include?(status)
+        ::HostStatus::Global::OK
+      elsif [APPROACHING_END_OF_SUPPORT].include?(status)
+        ::HostStatus::Global::WARN
+      elsif [SUPPORT_ENDED].include?(status)
+        ::HostStatus::Global::ERROR
+      else
+        ::HostStatus::Global::OK
+      end
+    end
+
+    def to_status(operatingsystem: nil)
+      self.class.to_status(operatingsystem: operatingsystem || self.host&.operatingsystem)
+    end
+
+    # this status is only relevant for RHEL
+    def relevant?(_options = {})
+      host&.operatingsystem&.rhel_eos_schedule
+    end
```

Comment:
If `relevant?` returns false, `refresh_host_statuses` will skip this status when refreshing. This is what causes it to remain "N/A" for non-RHEL hosts.

Reply:
Could we display this RHEL lifecycle information only for RHEL hosts? 
So nothing would be added for non-RHEL hosts at all.

Reply:
As far as I can tell, all registered statuses display for all hosts. This is what it looks like for a non-relevant host:

![image](https://github.com/Katello/katello/assets/22042343/daefeeb8-4ce2-4297-aadb-8f66cb1af5d7)

Since there's no way to register a host status only for certain hosts, I think this is the best we can do.